source/common/logger/ur_sinks.hpp

```diff
@@ -50,6 +45,14 @@ class Sink {
         flush_level = logger::Level::ERR;
     }
 
+    virtual void print(logger::Level level, std::string msg) {
```

Comment:
I think this might allocate  the string again. `const std::string &msg` ?

Reply:
Done